scripts/test_script.py

```diff
@@ -0,0 +1,409 @@
+"""
+Script for running tests of the cli as a whole via the command line
+
+Notes on usage:
+    - Replace DAFNI_SCRIPT_LOCATION and DAFNI_SNAPSHOT_SAVE_LOCATION as
+      necessary before executing (you can also assign them with environment
+      variables)
+    - Run on python command line e.g. python ./test/cli_test_script --help
+    - Should login just before running with --snapshot as otherwise token
+      could time out and get stuck
+"""
+
+import os
+import subprocess
+import tempfile
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Optional, Union
+
+import click
+
+home_dir = Path.home()
+
+# The cli executable is
+DAFNI_CLI_SCRIPT = os.getenv("DAFNI_CLI_SCRIPT") or f"{home_dir}/.local/bin/dafni"
+
+# Where to save snapshots
+DAFNI_SNAPSHOT_SAVE_LOCATION = (
```

Comment:
Maybe state somewhere that this needs to be an already existing folder

Reply:
Put it in the comment and in the dev notes.